data_safe_haven/external/api/azure_api.py

```diff
@@ -845,6 +845,11 @@ def remove_resource_group(self, resource_group_name: str) -> None:
             self.logger.debug(
                 f"Removing resource group [green]{resource_group_name}[/] if it exists...",
             )
+            if not resource_client.resource_groups.check_existence(resource_group_name):
+                self.logger.error(
+                    f"Resource group [green]{resource_group_name}[/] does not exist. There is no context infrastructure to teardown.",
+                )
+                return
```

Comment:
Is this an error? Would it be better as a warning or info?

Reply:
Wasn't sure, so put as error now to make it stand out. Question: in some places we are using bare print statements to give feedback to the user. Should we be using `self.logger.info/warning()`?

Reply:
I would probably make it a warning or info. I think error is really reserved for something you can't handle and causes the program to terminate.

Regarding print statements, it depends https://github.com/alan-turing-institute/data-safe-haven/issues/1891.
I don't think all output is log output.